mmedit/models/editors/stable_diffusion/stable_diffusion.py

```diff
@@ -0,0 +1,491 @@
+        image = image.cpu().permute(0, 2, 3, 1).float()
+        return image
+
+    def prepare_extra_step_kwargs(self, generator, eta):
+        """prepare extra kwargs for the scheduler step.
+
+        Args:
+            generator (torch.Generator):
+                generator for random functions.
+            eta (float):
+                eta (η) is only used with the DDIMScheduler,
+                it will be ignored for other schedulers.
+                eta corresponds to η in DDIM paper:
+                https://arxiv.org/abs/2010.02502
+                and should be between [0, 1]
+
+        Return:
+            extra_step_kwargs (dict):
+                dict contains 'generator' and 'eta'
+        """
+        accepts_eta = 'eta' in set(
+            inspect.signature(self.scheduler.step).parameters.keys())
+        extra_step_kwargs = {}
+        if accepts_eta:
+            extra_step_kwargs['eta'] = eta
+
+        # check if the scheduler accepts generator
+        accepts_generator = 'generator' in set(
+            inspect.signature(self.scheduler.step).parameters.keys())
+        if accepts_generator:
+            extra_step_kwargs['generator'] = generator
+        return extra_step_kwargs
+
+    def check_inputs(self, prompt, height, width):
+        """check whether inputs are in suitable format or not."""
+
+        if not isinstance(prompt, str) and not isinstance(prompt, list):
+            raise ValueError(f'`prompt` has to be of '
+                             f'type `str` or `list` but is {type(prompt)}')
+
+        if height % 8 != 0 or width % 8 != 0:
+            raise ValueError(f'`height` and `width` have to be divisible '
+                             f'by 8 but are {height} and {width}.')
+
+    def prepare_latents(self,
+                        batch_size,
+                        num_channels_latents,
+                        height,
+                        width,
+                        dtype,
+                        device,
+                        generator,
+                        latents=None):
+        """prepare latents for diffusion to run in latent space.
+
+        Args:
+            batch_size (int): batch size.
+            num_channels_latents (int): latent channel nums.
+            height (int): image height.
+            width (int): image width.
+            dtype (torch.dtype): float type.
+            device (torch.device): torch device.
+            generator (torch.Generator):
+                generator for random functions, defaults to None.
+            latents (torch.Tensor):
+                Pre-generated noisy latents, defaults to None.
+
+        Return:
+            latents (torch.Tensor): prepared latents.
+        """
+        shape = (batch_size, num_channels_latents,
+                 height // self.vae_scale_factor,
+                 width // self.vae_scale_factor)
+        if latents is None:
+            latents = torch.randn(
```

Comment:
Will `torch.randn` use random seed set in `set_random_seed` if `generator` is None?

Reply:
yes, set_random_seed will make seed be set globally.